crates/modelardb_compression/src/models/swing.rs

```diff
@@ -326,6 +350,26 @@ fn compute_slope_and_intercept(
     }
 }
 
+/// Computes numerator and denominator for Equation 6 
+/// in Section 3.2 of [Swing and Slide paper](https://dl.acm.org/doi/10.14778/1687627.1687645).
+/// It is a part of recording mechanism for finding the optimal slope with minimum MSE.
+fn compute_local_optimal_slope(
```

Comment:
Private functions and methods should be placed right after the last function or method that calls them, e.g., see `compute_slope_and_intercept()`. In this case the last caller is `compute_local_optimal_slope()`. If multiple private functions or methods are placed after a function or method they should be placed in the order they are called. So the one that is called first should be placed first, followed by the one that is called second, and so on.

Reply:
Since `fit_data_point()` is the last caller of `compute_slope_with_minimum_mse()`, `compute_slope_with_minimum_mse()` should be placed right after `fit_data_point()`.